### PR TITLE
[IOTDB-4431] Add QuickSort in TVList

### PR DESCRIPTION
**Pull request by LittleHealth**
Add quick sort algorithm;
1. add config, choose to use tim or quick
2. reformat: move clone, deserialize functions to abstract class IntTVList
3. fix the test

<hr>

This PR has:
- [ ] been self-reviewed.
    - [ ] concurrent read
    - [ ] concurrent write
    - [ ] concurrent read and write 
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. 
- [ ] added or updated version, __license__, or notice information
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious 
  for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold 
  for code coverage.
- [ ] added integration tests.
- [ ] been tested in a test IoTDB cluster.

<hr>

### CODE REVIEW

**Review by HeimingZ — COMMENTED** [review]
1. Use enumeration to denote different sort algorithm.
2. Serialize sort type and choose different implementations when deserializating.

**Review by HeimingZ — APPROVED** [review]
LGTM

File: server/src/assembly/resources/conf/iotdb-datanode.properties [hunk 1]
```diff
@@ -370,6 +370,12 @@ timestamp_precision=ms
 # Datatype: long
 # unseq_memtable_flush_check_interval_in_ms=600000
 
+# The sort algorithms used in the memtable's TVList
+# TIM: default tim sort,
+# QUICK: quick sort,
+# BACK: backward sort
+tvList_sort_algorithm=TIM
```
Comment (HTHou) [inline server/src/assembly/resources/conf/iotdb-datanode.properties:377]:
```suggestion
# tvlist_sort_algorithm=TIM
```

File: server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java [hunk 2]
```diff
@@ -1381,6 +1382,11 @@ private void loadTimedService(Properties properties) {
     if (unseqMemTableFlushCheckInterval > 0) {
       conf.setUnseqMemtableFlushCheckInterval(unseqMemTableFlushCheckInterval);
     }
+
+    conf.setTvListSortAlgorithm(
+        TVListSortAlgorithm.valueOf(
+            properties.getProperty(
+                "tvList_sort_algorithm", conf.getTvListSortAlgorithm().toString())));
```
Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:1389]:
```suggestion
                "tvlist_sort_algorithm", conf.getTvListSortAlgorithm().toString())));
```

Comment (HTHou) [inline server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:1389]: Why added this in `loadTimedService` method?

Comment (LittleHealth) [reply server/src/main/java/org/apache/iotdb/db/conf/IoTDBDescriptor.java:1389]: fixed, move it to the function loadProperties in line 395.

File: server/src/assembly/resources/conf/iotdb-datanode.properties [hunk 3]
```diff
@@ -370,6 +370,12 @@ timestamp_precision=ms
 # Datatype: long
 # unseq_memtable_flush_check_interval_in_ms=600000
 
+# The sort algorithms used in the memtable's TVList
+# TIM: default tim sort,
+# QUICK: quick sort,
+# BACKWARD: backward sort
+tvlist_sort_algorithm=TIM
```
Comment (HTHou) [inline server/src/assembly/resources/conf/iotdb-datanode.properties:377]:
```suggestion
# tvlist_sort_algorithm=TIM
```

**Pull request merged**